pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/ValidationManager.java

```diff
@@ -178,6 +190,22 @@ public void runValidation() {
     LOGGER.info("Validation completed");
   }
 
+  // If we add a new broker, we want to rebuild the broker resource.
+  void rebuildBrokerResourceWhenBrokerAdded(AbstractTableConfig tableConfig, IdealState brokerIdealState) {
+    String brokerTenantName = tableConfig.getTenantConfig().getBroker();
+    String tableName = tableConfig.getTableName();
+
+    Set<String> brokerTenantInstances = _pinotHelixResourceManager.getAllInstancesForBrokerTenant(brokerTenantName);
+    Set<String> idealStateBrokerInstances = brokerIdealState.getInstanceSet(tableName);
+
+    if(!idealStateBrokerInstances.equals(brokerTenantInstances)) {
+      _pinotHelixResourceManager.rebuildBrokerResourceFromHelixTags(tableName);
+      LOGGER.info("Rebuilding broker resource", tableName);
```

Comment:
It should be "Rebuilding broker resource for table {}" so that `tableName` gets filled in in your message.